dashboard/app/controllers/home_controller.rb

```diff
@@ -30,6 +32,13 @@ def set_locale
                     end
     # Query parameter for browser cache to be avoided and load new locale
     redirect_path = "#{redirect_path}?lang=#{params[:locale]}" if params[:locale]
+
+    redirect_uri = URI(redirect_path)
+    redirect_params = URI.decode_www_form(redirect_uri.query.to_s).to_h
+    redirect_params[Rack::GlobalEdition::REGION_KEY] = ge_region
```

Comment:
I suppose we need to only do this if `ge_region` has a value

Reply:
The idea is that when we set the param to nil, it resets the region

Reply:
Gotcha... so picking a language outside the region always redirects to a URL with `&ge_region` at the end.

so we can change the tests that change language then.

This test probably just needs to account for the empty `ge_region`:

```
test_redirects_cartoon_network_quick_links#RedirectsTest (1161.08s)
--
326 | Expected response to be a redirect to <https://test-studio.code.org/flappy/1?lang=ar-SA> but was a redirect to <https://test-studio.code.org/flappy/1?lang=ar-SA&ge_region>.
327 | Expected "https://test-studio.code.org/flappy/1?lang=ar-SA" to be === "https://test-studio.code.org/flappy/1?lang=ar-SA&ge_region".
328 | test/integration/redirects_test.rb:39:in `block in <class:RedirectsTest>'
329 | test/testing/setup_all_and_teardown_all.rb:36:in `run'
```

That seems fine. It's not a big deal to have these mechanical things be as mechanical as we need them to be.